purchases/src/main/kotlin/com/revenuecat/purchases/amazon/DefaultPurchasingServiceProvider.kt

```diff
@@ -18,7 +19,7 @@ internal class DefaultPurchasingServiceProvider : PurchasingServiceProvider {
     }
 
     override fun getUserData(): RequestId {
-        return PurchasingService.getUserData()
+        return PurchasingService.getUserData(UserDataRequest.newBuilder().setFetchLWAConsentStatus(true).build())
```

Comment:
I guess not but just to check, this shouldn't cause any behavior changes for users not using Quick Subscribe right? Just not sure why this is needed, if there is no behavior change 🤔

Reply:
I *think* it's possible the `lwaConsentStatus` field is not filled/requested unless this flag is set to true. This might be for backwards compatibility with older versions of the SDK.

https://developer.amazon.com/docs/in-app-purchasing/iap-quicksubscribe.html#implement-getuserdata-changes-and-integrate-login-with-amazon

Reply:
Feels like there should be a test making sure getUserData is called with this builder

Reply:
Hmm how could you test that? that's the direct call to Amazon's SDK.

Internally, the implementation of `getUserData` without parameters just calls the builder without setting the `setFetchLWAConsentStatus` flag.

<img width="880" alt="Screenshot 2024-06-18 at 12 31 22" src="https://github.com/RevenueCat/purchases-android/assets/1014118/add7cd91-4ba5-4f78-9c1d-83dc207759d5">

Reply:
I mean that we should check we are passing a builder that has  `setFetchLWAConsentStatus` set to true to `PurchasingService.getUserData` 

That way we find out if someone removes it and changes this function back to

```
 override fun getUserData(): RequestId {
        return PurchasingService.getUserData()
 }
 ```

Reply:
You can potentially mock static/builder methods, similar to how we do for Google here: https://github.com/RevenueCat/purchases-android/blob/5deff6cbebe4712ef3633053ceb6d217179f4fa6/purchases/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt#L325

Reply:
Thanks for sending that Toni. I thought my original comment was confusing and could mean "test the insights of Amazon to make sure the method is called with the right parameters". 

But yes I'm meaning make sure the builder passed has `setFetchLWAConsentStatus` set. I thought you could mock `PurchasingService` and capture the parameter passed to `getUserData`, but just checked and looks like `UserDataRequest` doesn't expose `fetchLWAConsentStatus`, which makes it a bit more complicated...

Feel free to ignore if it's too complicated, but I think you should be able to test it by doing what Toni mentions and mocking the static

```
 mockkStatic(UserDataRequest::class)
 val mockBuilder = mockk<UserDataRequest.Builder>(relaxed = true)
  every {
      UserDataRequest.newBuilder()
  } returns mockBuilder
```

Reply:
I get it now! Added a test specifically for testing `DefaultPurchasingServiceProvider`. Testing it directly seemed like the easiest way and didn't feel like creating a new file for it was needed but let me know if you'd improve the test in some way!